why-polars-for-data-cleaning/index.qmd

```diff
@@ -0,0 +1,185 @@
+---
+title: "Why use Polars for data cleaning?"
+description: "Key to all data storage is good data quality. This is often overlooked as it can be labour intensive, but it is essential for good data analysis. This post explains why we chose to use Polars for data cleaning."
+date: "2024-12-04"
+categories:
+- data cleaning
+- pre-processing data
+---
+
+::: content-hidden
+Use other decision posts as inspiration to writing these. Leave the
+content-hidden sections in the text for future reference.
+:::
+
+## Context and problem statement
+
+::: content-hidden
+State the context and some background on the issue, then write a
+statement in the form of a question for the problem.
+:::
+
+When working with data analysis, data is rarely ready to be used 'as
+is'. There will always be things that are either very wrong (like text
+in what should be a number field), or slightly wrong (like a single
+missing value, or an outlier, which may or may not be a data entry
+error). In Sprout we are not worried about the single missing values or
+the outliers, but we are concerned with the data that contradicts the
+definition of the resource (eg. TBC in a date field). There are a lot of
+tools that can be used to both analyse a data set, and to clean it,
+either as part of the analysis, or as part of the initial check before
+the data is transformed and stored. The checking of the data prior to
+inclusion into Sprout is done within Sprout itself, this work will
+result in a list of errors which the user will need to handle before
+they can load the data into Sprout.
+
+> Which tool should we use for cleaning and processing data before
+> loading it into Sprout?
+
+## Decision drivers
+
+::: content-hidden
+List some reasons for why we need to make this decision and what things
+have arisen that impact work.
+:::
+
+-   We need a tool that can clean data and is easy to integrate with our
+    current workflow.
```

Comment:
Could you expand on this? What do you mean by "integrate with our workflow"?

Reply:
We are building things in python, so if we expect our users to know a language it would be python and not for instance R. So it needs to be easy for us to pick up and use. I think is how I was thinking of it. I'd be happy to take that bit out.

Reply:
No need to take it out, just make it clearer by adding what you wrote here! 😁 That it needs to have a Python package available.